js/core/src/tracing/instrumentation.ts

```diff
@@ -118,10 +122,32 @@ export async function runInNewSpan<T>(
           opts.labels
         );
 
+        const lastGenkitSpanId =
+          getAsyncContext().getStore<string>(lastGenkitSpanIdAlsKey) || '';
+        const isGenkitSpan = !!opts.labels?.[SPAN_TYPE_ATTR];
+        if (isGenkitSpan && parentStep) {
+          const parentIsGenkit = !!parentStep.labels?.[SPAN_TYPE_ATTR];
+          if (!parentIsGenkit && lastGenkitSpanId) {
+            otSpan.setAttribute(
+              ATTR_PREFIX + ':lastKnownParentSpanId',
+              lastGenkitSpanId
+            );
+          }
+        }
+
         const output = await getAsyncContext().run(
           spanMetadataAlsKey,
           spanContext,
-          () => fn(opts.metadata, otSpan, isInRoot)
+          () => {
+            if (isGenkitSpan) {
+              return getAsyncContext().run(
+                lastGenkitSpanIdAlsKey,
+                otSpan.spanContext().spanId,
+                () => fn(opts.metadata, otSpan, isInRoot)
+              );
```

Comment:
how is this different from what we already put in `spanMetadataAlsKey`? why do you need a separate ALS?